alacritty/src/display.rs

```diff
@@ -592,9 +577,28 @@ impl Display {
             self.renderer.draw_rects(&size_info, rects);
         }
 
-        self.draw_search(config, &size_info, message_bar_lines, search_regex);
         self.draw_render_timer(config, &size_info);
 
+        // Handle search bar rendering.
+        let ime_position = match search_state.regex() {
+            Some(regex) => {
+                let search_label = match search_state.direction() {
+                    Direction::Right => FORWARD_SEARCH_LABEL,
+                    Direction::Left => BACKWARD_SEARCH_LABEL,
+                };
+
+                let regex = Self::format_search(&size_info, regex, search_label);
+
+                self.draw_search(config, &size_info, message_bar_lines, &regex);
+
+                Point::new(size_info.lines() - 1, Column(regex.len() - 1))
```

Comment:
In theory it would be possible to do all this in the `draw_search` method, but I don't like returning the ime position from the `draw_search` function, that doesn't make a lot of sense to me. And doing more than just compute the ime position in a function that is meant to compute it wouldn't be a great idea either. So I think this is the best approach.